website_updater.py

```diff
@@ -95,21 +110,44 @@ def request_responses(data_url: str) -> List[Dict[str, str]]:
     return responses
 
 
-def cleanup_repo(repo: str, branch_name: str):
-    execute_cmd(["git", "restore", "."], repo)
-    execute_cmd(["git", "checkout", "master"], repo)
-    execute_cmd(["git", "reset", "--hard", "HEAD"], repo)
-    execute_cmd(["git", "pull"], repo)
-    execute_cmd(["git", "checkout", "-b", branch_name], repo)
+def cleanup_repo(repo: str, branch_name: str, dry_run: bool):
+    execute_cmd(["git", "restore", "."], repo, dry_run)
+    execute_cmd(["git", "checkout", "master"], repo, dry_run)
+    execute_cmd(["git", "reset", "--hard", "HEAD"], repo, dry_run)
+    execute_cmd(["git", "pull"], repo, dry_run)
+    execute_cmd(["git", "checkout", "-b", branch_name], repo, dry_run)
 
 
-def create_pr(repo, branch_name, now):
+def create_pr(repo: str, branch_name: str, now: datetime.datetime, dry_run: bool):
     title = f"chore: update to {now.strftime('%Y-%m-%d %H:%M')}"
     body = f"Automatic update of the website posts. Data retrieval timestamp: {now}"
-    execute_cmd(["git", "commit", "-a", "-m", title], repo)
-    execute_cmd(["git", "push", "-u", "origin", branch_name], repo)
-    execute_cmd(["gh", "pr", "create", "--title", title, "--body", body], cwd=repo)
+    execute_cmd(["git", "commit", "-a", "-m", title], repo, dry_run)
+    execute_cmd(["git", "push", "-u", "origin", branch_name], repo, dry_run)
+    execute_cmd(["gh", "pr", "create", "--title", title, "--body", body], repo, dry_run)
+
+
+def calc_countries(responses: List[dict]):
+    return len(set([val["country_en"] for val in responses]))
```

Comment:
nit: we can use set comprehension directly:
```suggestion
    return len({val["country_en"] for val in responses})
```
or at least skip unpacking to an intermediate list, if we want to keep `set` explicit:
```suggestion
    return len(set(val["country_en"] for val in responses))
```

Reply:
+1 to the 2nd option.

Reply:
Yup, let's go with option 2.